src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDag.cs

```diff
@@ -196,6 +194,31 @@ BoundDecisionDagNode makeReplacement(BoundDecisionDagNode dag, Func<BoundDecisio
             }
         }
 
+        /// <summary>
+        /// To facilitate subsumption-checking for list-patterns, some nodes may be inserted into the dag during construction.
+        /// We will remove such nodes immediately after as these are not important in lowering or any other analysis.
+        /// </summary>
+        public BoundDecisionDag RemoveSubsumptionOnlyNodes()
+        {
+            return Rewrite(static (node, replacement) =>
```

Comment:
To avoid an entire rewrite, we can move this check to `ComputeBoundDecisionDagNodes`. Will try it in the next iteration.